extensions/dim-rename-kafka8/src/main/java/io/druid/server/namespace/KafkaExtractionManager.java

```diff
@@ -0,0 +1,239 @@
+import kafka.consumer.ConsumerIterator;
+import kafka.consumer.KafkaStream;
+import kafka.consumer.Whitelist;
+import kafka.javaapi.consumer.ConsumerConnector;
+import kafka.message.MessageAndMetadata;
+import kafka.serializer.Decoder;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Pattern;
+
+/**
+ *
+ */
+@ManageLifecycle
+public class KafkaExtractionManager
+{
+  private static final Logger log = new Logger(KafkaExtractionManager.class);
+
+  private final Properties kafkaProperties;
+  private final ConcurrentMap<String, Function<String, String>> fnCache;
+  private final Collection<ListenableFuture<?>> futures = new ConcurrentLinkedQueue<>();
+  private final ListeningExecutorService executorService = MoreExecutors.listeningDecorator(
+      Executors.newCachedThreadPool(
+          new ThreadFactoryBuilder()
+              .setNameFormat("kafka-rename-consumer-%d")
+              .setDaemon(true)
+              .setPriority(Thread.MIN_PRIORITY)
+              .build()
+      )
+  );
+  private final NamespaceExtractionCacheManager cacheManager;
+  private final AtomicInteger startedPipes = new AtomicInteger(0);
+  private final AtomicLong events = new AtomicLong(0l);
+
+  // Bindings in KafkaExtractionNamespaceModule
+  @Inject
+  public KafkaExtractionManager(
+      @Named("renameKafkaProperties") Properties kafkaProperties,
+      NamespaceExtractionCacheManager cacheManager,
+      @Named("io.druid.server.namespace.NamespacedExtractionModule")
+      ConcurrentMap<String, Function<String, String>> fnCache
+  )
+  {
+    this.kafkaProperties = new Properties();
+    this.fnCache = fnCache;
+    if (kafkaProperties != null) {
+      if (kafkaProperties.containsKey("group.id")) {
+        throw new IAE(
+            "Cannot set kafka property [group.id]. Property is randomly generated for you. Found [%s]",
+            kafkaProperties.getProperty("group.id")
+        );
+      }
+      if (kafkaProperties.containsKey("auto.offset.reset")) {
+        throw new IAE(
+            "Cannot set kafka property [auto.offset.reset]. Property will be forced to [smallest]. Found [%s]",
+            kafkaProperties.getProperty("auto.offset.reset")
+        );
+      }
+      this.kafkaProperties.putAll(kafkaProperties);
+    }
+    if (!this.kafkaProperties.containsKey("zookeeper.connect")) {
+      this.kafkaProperties.put("zookeeper.connect", "localhost:2181/kafka");
+    }
+    // Enable publish-subscribe
+    this.kafkaProperties.setProperty("auto.offset.reset", "smallest");
+    this.cacheManager = cacheManager;
+  }
+
+  public Integer getBackgroundTaskCount()
```

Comment:
why not simply return an int here and save the auto boxing-inboxing , most of the clients of this method will probably just need int. Conventionally, boxed primitives are used in exception cases where unboxed ones can't just work.
if you agree with this , then probly you can make the change throughout the PR